CI/include/build_support_windows_cross.sh

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+
+##############################################################################
+# Windows cross-compile support functions
+##############################################################################
+#
+# This script file can be included in build scripts for Windows.
+#
+##############################################################################
+
+# Setup build environment
+
+PARALLELISM="$(nproc)"
+CI_WORKFLOW="${CHECKOUT_DIR}/.github/workflows/windows_deps.yml"
+
+if [ "${TERM-}" -a -z "${CI}" ]; then
+    COLOR_RED=$(/usr/bin/tput setaf 1)
+    COLOR_GREEN=$(/usr/bin/tput setaf 2)
+    COLOR_BLUE=$(/usr/bin/tput setaf 4)
+    COLOR_ORANGE=$(/usr/bin/tput setaf 3)
+    COLOR_RESET=$(/usr/bin/tput sgr0)
+else
+    COLOR_RED=""
+    COLOR_GREEN=""
+    COLOR_BLUE=""
+    COLOR_ORANGE=""
+    COLOR_RESET=""
+fi
+
+## DEFINE UTILITIES ##
+install_tools() {
+    sudo apt ${QUIET:+--quiet} -y install automake cmake curl git libtool mingw-w64 mingw-w64-tools pkg-config wget
+
+    curl -L -O https://www.nasm.us/pub/nasm/releasebuilds/2.15.01/nasm-2.15.01.tar.xz
+    tar -xf nasm-2.15.01.tar.xz
```

Comment:
It's hash checked in the build script itself because we're just using prebuilt versions of NASM and we need both 32-bit and 64-bit, for which the hashes differ.

Reply:
Hash should now be checked for this download.